reframe/core/pipeline.py

```diff
@@ -734,10 +738,16 @@ def __init__(self):
         pass
 
     @classmethod
-    def __init_subclass__(cls, *, special=False, **kwargs):
+    def __init_subclass__(cls, *, special=False, base_test=False, **kwargs):
         super().__init_subclass__(**kwargs)
         cls._rfm_special_test = special
 
+        # Insert the base test path as the prefix if the class is marked
+        # as a base test.
+        if base_test:
+            cls._rfm_base_test_prefix = os.path.abspath(
+                os.path.dirname(inspect.getfile(cls)))
```

Comment:
Here you need also the `is_interactive()` check, so I suggest moving the setting of the prefix into a separate function.

Reply:
What about just adding this attribute in any case, and then using the `is_interactive` check from the comment below?